src/QuantumComputation.cpp

```diff
@@ -632,7 +635,7 @@ void QuantumComputation::dump(const std::string& filename) {
   }
 }
 
-void QuantumComputation::dumpOpenQASM(std::ostream& of) {
+void QuantumComputation::dumpOpenQASM(std::ostream& of, bool openQASM3) {
```

Comment:
It feels to me as if it would be better to split this into two methods and extract the common functionality (such as consolidating the registers) into another separate function.